source/digits_hits/include/GateJPETGammaPolarizationActor.hh

```diff
@@ -0,0 +1,302 @@
+	 * */
+	void SetDiagnosticFile(G4String diag_file_name);
+
+	/** Set filtration range for phi
+	 * @param: value - value to set - depends on what is mode you have to input other value (for more look to function body)
+	 * @param: mode - mode inform how to interpret value
+	*/
+	void SetPhiFlagParameter(G4double value, GateJPETGammaPolarizationActor::PhiThetaValueMode mode);
+
+	/** Set filtration range for theta
+	 * @param: value - value to set - depends on what is mode you have to input other value (for more look to function body)
+	 * @param: mode - mode inform how to interpret value
+	*/
+	void SetThetaFlagParameter(G4double value, GateJPETGammaPolarizationActor::PhiThetaValueMode mode);
+
+	/** Set filtration range for phi
+	 * @param: setAsPlus - if is necessary to save phi angle as only positive value this flag must be set as true
+	*/
+	void SetAllPhiAsNoNegative(G4bool setAsPlus);
+
+	/** Set generation of histogram as (dis)enable.
+	 * By default generation is enable;
+	 * @param: enable - set false to disable histograms generations
+	*/
+	void SetHistogramsEnable(G4bool enable);
+
+	/** Set generation of tests branch (for polarization) as (dis)enable.
+	 * By default generation is enable;
+	 * @param: enable - set false to disable tests branch generations
+	*/
+	void SetTestsBranchEnable(G4bool enable);
+
+	/** Set angle precision per bin in histograms - this mean that each bin has width equal angle_precison
+	 * By default precision is 2 degree.
+	 * @param: angle_precision - bin width (in degrees)
+	 * @exceptions: If value is to huge, what mean angle_precision >= 180.0 deg then this value is ignored.
+	*/
+	void SetAnglePrecisionPerBinInHistograms(G4double angle_precision);
+
+	/** Set (dis)enable saving data ony for desired number of particles per event which scattered.
+	 * The purpose of this to reduce number of data generated during simulation.
+	 * @param: enable - set true to use this, false - otherwise
+	 * */
+	void SetSaveOnlyWhenTheDesiredNumberOfParticlesHasScatteredFromEventEnable(G4bool enable);
+
+	/** Set desired number of particles scattered per event.
+	 * @param: particles_number - if number of scattered per event is equal particles_number then data will be saved
+	 * */
+	void SetDesiredNumberOfParticlesScatteredPerEvent(G4int particles_number);
+
+private: //Functions
+	/** Function generate perpendicular vector to vector 'a'
+	 * @param: a - vector for which we create perpendicular one
+	 * @return: perpendicular vector to 'a'
+	*/
+	G4ThreeVector SetPerpendicularVector(G4ThreeVector& a);
+
+	/* Special function just for this class.
+	* Extract values setted by Set functions.
+	* */
+	void StandardExtractFunction(const G4Step *step);
+
+	/** Function generate dsigma/dOmega as function of theta.
+	 * For this this function generate histogram which is histogram dsigma/dOgema divided by sin(theta) histogram.
+	 * */
+	void GenerateRealCrossSectionOfTheta();
+
+	/** Function normalize major of histograms
```

Comment:
probably you should give a list of histos here? Or maybe explicitely pass a list of histos to normalize

Reply:
But why? Not all histos are required to be normalised plus creating list force me to change logic of code - generaly speking I can use list of uniq_ptr's and in this case remember about **delete** is unnecessery.

Reply:
Ok you can leave it as it is.
In general I prefer to know what the function does. From the description you have no idea which histograms will be normalize and which not.